Functions/ImageProcessing/xASL_im_Lesion2Mask.m

```diff
@@ -198,17 +198,18 @@
     OutIm{iIm} = xASL_vis_CreateVisualFig(x, {x.P.Pop_Path_rT1 VisualizeImage==iIm}, [], [0.75 0.35], [], {x.S.gray x.S.colors_ROI{iIm}});
 end
 
-% Initialize final image
-OutImFinal = zeros(size(OutIm{1}));
-OutImFinal(OutIm{1}==OutIm{2}) = OutImFinal(OutIm{1}==OutIm{2});
+% Initialize final image - make sure it has the proper dimensions when an individual image is returned empty
+OutImFinal = zeros(max(cellfun(@(x) size(x,1),OutIm)), max(cellfun(@(x) size(x,2),OutIm)), max(cellfun(@(x) size(x,3),OutIm)));
 
 for iIm=1:length(OutIm)
-    % Create masks with non-grey voxels (i.e. the overlay)
-    OutImMask{iIm} = OutIm{iIm}(:,:,1)~=OutIm{iIm}(:,:,2) | OutIm{iIm}(:,:,2)~=OutIm{iIm}(:,:,3)  | OutIm{iIm}(:,:,1)~=OutIm{iIm}(:,:,3);
-    OutImMask{iIm} = repmat(OutImMask{iIm}, [1 1 3]);
-    
-    % Add the overlay from each mask
-    OutImFinal(OutImMask{iIm}) = OutIm{iIm}(OutImMask{iIm});
+	if isequal(size(OutImFinal), size(OutIm{iIm}))
```

Comment:
What do we do if they don't have equal size? Now they are skipped, at least we should provide a warning to the user?

Are you sure this is correct usage? The user needs to provide Lesion_T1.nii (same space as the T1.nii). If this is not the case, we should create an error earlier in the pipeline! The user will otherwise just continue the code, even if we put a warning here, they won't read all the output.

Plus, if you create this error earlier, we don't need these changes?